packages/cli/src/lib/setups/i18n/replacers.ts

```diff
@@ -342,3 +342,192 @@ async function findEntryFile({
 
   return {filepath, astType};
 }
+
+type FilePath = {
+  rootDirectory: string;
+  path: string;
+  filename: string;
+};
+
+async function findFile({rootDirectory, path, filename}: FilePath) {
+  if (!fileExists(joinPath(rootDirectory, path, filename))) return {};
+
+  const {filepath, astType} = await findFileWithExtension(
+    joinPath(rootDirectory, path),
+    filename,
+  );
+
+  return {filepath, astType};
+}
+
+export async function replaceI18nCartPath(
+  {rootDirectory}: I18nSetupConfig,
+  formatConfig: FormatOptions,
+) {
+  await Promise.all([
+    replaceI18nRootFile({rootDirectory}, formatConfig),
+    replaceI18nCartPathForFile(
+      {
+        rootDirectory,
+        path: 'app/components',
+        filename: 'Cart',
+      },
+      formatConfig,
+    ),
+    replaceI18nCartPathForFile(
+      {
+        rootDirectory,
+        path: 'app/components',
+        filename: 'AddToCartButton',
+      },
+      formatConfig,
+    ),
+  ]);
+}
+
+async function replaceI18nRootFile(
+  {rootDirectory}: I18nSetupConfig,
+  formatConfig: FormatOptions,
+) {
+  // Add selectedLocale to root loader
+  const {filepath, astType} = await findFile({
+    rootDirectory,
+    path: 'app',
+    filename: 'root',
+  });
+
+  if (!filepath || !astType) {
+    throw new AbortError('Could not find a root file');
+  }
+
+  await replaceFileContent(filepath, formatConfig, async (content) => {
+    const astGrep = await importLangAstGrep(astType);
+    var root = astGrep.parse(content).root();
+    const loaderReturnNode = root.find({
+      rule: {
+        kind: 'shorthand_property_identifier',
+        regex: 'publicStoreDomain',
+      },
+    });
+
+    if (loaderReturnNode) {
+      const {end} = loaderReturnNode.range();
+      content =
+        content.slice(0, end.index + 1) +
+        'selectedLocale: storefront.i18n,' +
+        content.slice(end.index + 1);
+    }
+
+    root = astGrep.parse(content).root();
+    const loaderHookNode = root.find({
+      rule: {
+        kind: 'export_statement',
+        regex: 'useRootLoaderData',
+      },
+    });
+
+    if (loaderHookNode) {
+      const {end} = loaderHookNode.range();
+      content =
+        content.slice(0, end.index + 1) +
+        `
+          const DEFAULT_LOCALE = {
+            pathPrefix: '',
+            language: 'EN',
+            country: 'US',
+          };
+
+          export function usePrefixPathWithLocale(path: string) {
+            const rootData = useRootLoaderData();
+            const selectedLocale = rootData?.selectedLocale ?? DEFAULT_LOCALE;
+
+            return \`\${selectedLocale.pathPrefix}\${
+              path.startsWith('/') ? path : '/' + path
+            }\`;
+          }
+          ` +
+        content.slice(end.index + 1);
+    }
```

Comment:
I think we should move this to a separate file in `app/lib/i18n.ts` or `app/lib/locale.ts`. I also want to move the other i18n setup logic (`getLocaleFromRequest`) from `server.ts` to lib but unsure when we can do this.

Finding `useRootLoaderData` is not ensured (the user can modify that) and it adds some clutter to the root file 🤔 
Therefore, even if we move it, perhaps using `useMatches()[0]` is still better than `useRootLoaderData`.

Also, I suggest we export `useCartPath` from this same file where the user can easily configure `/cart` only once. Then we just need to update the other files with `route={useCartPath()}` or similar.

In fact, if we replace the component files first, we can find a record the cart route they use (e.g. `/cart` or `/shipping-cart`, etc.) and use that information as the value in `useCartPath`.

---

Aside from that, our existing i18n setup logic is adding already a `defaultLocale` variable. We should unify these two somehow but perhaps in a different PR is better?